addons/mail/models/mail_template.py

```diff
@@ -361,7 +361,8 @@ def get_email_template(self, res_ids):
             return results
         self.ensure_one()
 
-        langs = self.render_template(self.lang, self.model, res_ids)
+        lang = self.env.context.get('template_preview_lang', self.lang)
+        langs = self.render_template(lang, self.model, res_ids)
```

Comment:
Well I don't think it is the correct way of doing it. Previous code was rendering the lang field of the template, which could be a jinja template. Here we want to bypass this process and force a lang. I think it is better to a/ check for context key, then b/ if not found, render the template lang field.